project_RL/monte_carlo/train.py

```diff
@@ -0,0 +1,111 @@
+from project_RL.monte_carlo.monte_carlo_agent import MonteCarlo
+from gym_minigrid.wrappers import *
+from time import time
+
+
+def train(env, hyperparameters):
+    """ Train a sarsa lambda agent in the requested environment
+
+    Arguments:
+        hyperparameters dictionary containing:
+            - env_name
+            - n zero value
+    """
+    agent = MonteCarlo(env, hyperparameters['n_zero'])
+    agent.init_eligibility_table()
+
+    # create log file, add hyperparameters into it
+    env_name = hyperparameters['env_name']
+    log_filename = f'log_{env_name}_{time()}.csv'
+    with open(log_filename, 'a') as f:
+        f.write('\n'.join(map(','.join, {str(key): str(value) for key, value in hyperparameters.items()}.items())))
+        f.write('\n')
+        # write csv header
+        f.write('Episode,Step,Total Reward,q_value_table_length\n')
+
+    
+    # initialise variables for plotting purpose
+    step = 0
+
+    for episode in range(int(1e4)):
+        # reset environment before each episode
+        total_reward = 0.0
+
+        # to update our agent after each episode
+        states, actions, rewards = [], [], []
+
+        observation = env.reset()
+        state = parse_observation_to_state(observation)
+        action = agent.get_new_action_e_greedly(state)
+        done = False
+

```

Comment:
Acho que precisa chamar a função `init_eligibility_table` do agente no início de cada episódio, pois a contagem das visitas aos estado-ação é dentro de cada episódio se não me engano.

Reply:
A média é calculada dado o número da contagem (estado-ação) de todos os episódios.

Reply:
Sim, está certo.